python/cudf_polars/cudf_polars/experimental/expressions.py

```diff
@@ -168,6 +169,19 @@ def _decompose_unique(
         names=names,
     )
     (column,) = columns
+
+    assert config_options.executor.name == "streaming", (
+        "'in-memory' executor not supported in '_decompose_unique'"
+    )
+
+    cardinality: float | None = None
+    if cardinality_factor := {
+        v
```

Comment:
Should we add the same check here as in https://github.com/rapidsai/cudf/pull/18725/files#diff-91044f2adf9fac866e8511f93376be08a1954f572c9b314cf3b0910022a98082R160 ?